pkg/flow/module.go

```diff
@@ -70,6 +120,7 @@ func (c *module) LoadConfig(config []byte, args map[string]any) error {
 //
 // Run blocks until the provided context is canceled.
 func (c *module) Run(ctx context.Context) {
+	defer c.o.parent.removeID(c.o.ID)
```

Comment:
Can you add documentation to `component.Module.Run` that says the ID of a module won't be freed for re-use until the module stops running?

I want to use that documentation to make sure callers know that modules must _always_ be run, otherwise their IDs will be registered forever.